src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.LocalFunctionReferenceRewriter.cs

```diff
@@ -82,10 +82,8 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
                         out method,
                         ref arguments);
 
-                    var newType = _lambdaRewriter.VisitType(node.Type);
```

Comment:
>var newType = _lambdaRewriter.VisitType(node.Type); [](start = 20, length = 51)

Is visiting ```node.Type``` no longer necessary? Consider adding an assert that ```node.Type == method.ReturnType```. And for other similar places in this file.